internal/services/keyvault/key_vault_certificate_contacts_resource.go

```diff
@@ -158,24 +169,43 @@ func (r KeyVaultCertificateContactsResource) Read() sdk.ResourceFunc {
 			subscriptionResourceId := commonids.NewSubscriptionID(subscriptionId)
 			keyVaultIdRaw, err := vaultClient.KeyVaultIDFromBaseUrl(ctx, subscriptionResourceId, id.KeyVaultBaseUrl)
 			if err != nil {
-				return fmt.Errorf("retrieving resource ID of the Key Vault at URL %s: %+v", id.KeyVaultBaseUrl, err)
+				return fmt.Errorf("retrieving resource ID of the Key Vault from Key Vault Base URL %q in %s: %+v", id.KeyVaultBaseUrl, subscriptionResourceId, err)
 			}
+
 			if keyVaultIdRaw == nil {
-				metadata.Logger.Infof("Unable to determine the Resource ID for the Key Vault at URL %s - removing from state!", id.KeyVaultBaseUrl)
+				metadata.Logger.Infof("[READ] unable to retrieve Key Vault resource ID from Key Vault Base URL %q in %s - removing from state", id.KeyVaultBaseUrl, subscriptionResourceId)
 				return metadata.MarkAsGone(id)
 			}
+
 			keyVaultId, err := commonids.ParseKeyVaultID(*keyVaultIdRaw)
 			if err != nil {
 				return fmt.Errorf("parsing Key Vault ID: %+v", err)
 			}
 
 			existing, err := client.GetCertificateContacts(ctx, id.KeyVaultBaseUrl)
 			if err != nil {
+				// The contacts may or may not have changed but we do not have access to them due
+				// to the key vault public network access has been disabled... should just exit
+				// since we have no access to dataplane... This will preserve the contacts
+				// in the state file so once we do regain access to the dataplane we will be able
+				// to diff the resources contacts with the key vaults contacts...
+				if utils.ResponseWasForbidden(existing.Response) {
+					metadata.Logger.Infof("[READ] unable to enumerate key vault certificate contacts at URL %s in %s - ignore error", id.KeyVaultBaseUrl, subscriptionResourceId)
+					return nil
+				}
```

Comment:
I still think in this case we shall just error out, as it is similar to a mgmt plane resource that exists but you don't have permission to access it?

Reply:
As stated above, that is prolly more consistent with standard provider behavior, updated to return and error instead of swallowing the error.